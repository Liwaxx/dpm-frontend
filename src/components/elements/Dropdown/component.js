import React from 'react';
import PropTypes from 'prop-types';
import Dropdown from 'react-dropdown';
import 'react-dropdown/style.css';

export default class Component extends React.Component {
  render() {
    const options = [
      { value: 'week', label: 'Bulan/Minggu' },
      { value: 'month', label: 'Bulan/Tahun' }
    ];
    const defaultOption = options[0];
    
    return (
      <Dropdown
        options={options}
        value={defaultOption}
        name="dropdown" 
        onChange={this.props.onChange}
      />
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object,
  onChange: PropTypes.func,
};