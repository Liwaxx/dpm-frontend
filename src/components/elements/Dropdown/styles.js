const styles = ({
  top: {
    fontFamily: 'roboto',
    fontSize: '10px',
  },
  drop: {
    width: '10px',
  }
});
export default styles;