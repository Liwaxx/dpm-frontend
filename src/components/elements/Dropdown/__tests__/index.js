import React from 'react';
import renderer from 'react-test-renderer';
import Dropdown from '../index';

describe('Dropdown', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Dropdown />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});