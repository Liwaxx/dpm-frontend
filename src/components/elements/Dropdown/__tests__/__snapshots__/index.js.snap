// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
<div
  className="Dropdown-root"
>
  <div
    className="Dropdown-control"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="Dropdown-placeholder is-selected"
    >
      Bulan/Minggu
    </div>
    <div
      className="Dropdown-arrow-wrapper"
    >
      <span
        className="Dropdown-arrow"
      />
    </div>
  </div>
</div>
`;
